src/stella_vslam_ros.cc

```diff
@@ -204,8 +209,8 @@ void system::init_pose_callback(
                                    * initialpose_affine * robot_base_frame_to_camera_affine)
                                       .matrix();
 
-    const Eigen::Vector3d normal_vector = (Eigen::Vector3d() << 0., 1., 0.).finished();
-    if (!slam_->relocalize_by_pose_2d(cam_pose_cv, normal_vector)) {
+    const Eigen::Vector3d normal_vector = (Eigen::Vector3d() << 0., 0., 1.).finished();
+    if (!slam_->relocalize_by_pose_2d(cam_pose_cv, ros_map_to_cv_map_affine_.inverse().linear() * normal_vector)) {
```

Comment:
I'm not sure this is correct, unless I'm misinterpreting your intention with ros_map_to_cv_map_affine_.
My assumption was you want it to add a customizable offset between desired map frame and the initial pose when creating the map.
Do you instead intend users to specify the entire transform from base_link to camera_optical_link (on top of the desired map offset)?
- If so then I think this code is correct, but I think publish_pose then is invalid.
- If not, I think this part of the code probably needs a different transform for the normal vector, but I'm getting confused and can easily say what exactly.

Doesn't cam_pose_cv also need a transform including ros_map_to_map_affine in some way?

Reply:
Fixed in 316a220.